pkg/build/admission/defaults/admission_test.go

```diff
@@ -130,3 +132,137 @@ func TestIncrementalDefaults(t *testing.T) {
 	}
 
 }
+
+func TestLabelDefaults(t *testing.T) {
+	tests := []struct {
+		buildLabels   []buildapi.ImageLabel
+		defaultLabels []buildapi.ImageLabel
+		expected      []buildapi.ImageLabel
+	}{
+		{
+			buildLabels:   nil,
+			defaultLabels: nil,
+			expected:      nil,
+		},
+		{
+			buildLabels: nil,
+			defaultLabels: []buildapi.ImageLabel{
+				{
+					Name:  "distribution-scope",
+					Value: "private",
+				},
+				{
+					Name:  "changelog-url",
+					Value: "file:///dev/null",
+				},
+			},
+			expected: []buildapi.ImageLabel{
+				{
+					Name:  "distribution-scope",
+					Value: "private",
+				},
+				{
+					Name:  "changelog-url",
+					Value: "file:///dev/null",
+				},
+			},
+		},
+		{
+			buildLabels: []buildapi.ImageLabel{
+				{
+					Name:  "distribution-scope",
+					Value: "private",
+				},
+				{
+					Name:  "changelog-url",
+					Value: "file:///dev/null",
+				},
+			},
+			defaultLabels: nil,
+			expected: []buildapi.ImageLabel{
+				{
+					Name:  "distribution-scope",
+					Value: "private",
+				},
+				{
+					Name:  "changelog-url",
+					Value: "file:///dev/null",
+				},
+			},
+		},
+		{
+			buildLabels: []buildapi.ImageLabel{
+				{
+					Name:  "distribution-scope",
+					Value: "private",
+				},
+			},
+			defaultLabels: []buildapi.ImageLabel{
+				{
+					Name:  "distribution-scope",
+					Value: "public",
+				},
+				{
+					Name:  "changelog-url",
+					Value: "file:///dev/null",
+				},
+			},
+			expected: []buildapi.ImageLabel{
+				{
+					Name:  "distribution-scope",
+					Value: "private",
+				},
+				{
+					Name:  "changelog-url",
+					Value: "file:///dev/null",
+				},
+			},
+		},
+		{
+			buildLabels: []buildapi.ImageLabel{
+				{
+					Name:  "distribution-scope",
+					Value: "private",
+				},
+			},
+			defaultLabels: []buildapi.ImageLabel{
+				{
+					Name:  "changelog-url",
+					Value: "file:///dev/null",
+				},
+			},
+			expected: []buildapi.ImageLabel{
+				{
+					Name:  "distribution-scope",
+					Value: "private",
+				},
+				{
+					Name:  "changelog-url",
+					Value: "file:///dev/null",
+				},
+			},
+		},
```

Comment:
needs a test where the build object has no labels so you can confirm you can successfully add labels to a build definition that has an empty label field.